srcopsmetrics/github_knowledge.py

```diff
@@ -336,6 +346,63 @@ def store_pull_request(self, pull_request: PullRequest, results: Dict[str, Dict[
             "requested_reviewers": self.extract_pull_request_review_requests(pull_request),
         }
 
+    def store_content_file(self, file_content: ContentFile, results: Dict[str, Dict[str, Any]]):
+        """Analyse pull request and save its desired features to results.
+
+        Arguments:
+            file_content {ContentFile} -- Content File type to be stored.
+            results {Dict[str, Dict[str, Any]]} -- dictionary where all the currently
+                                                PRs are stored and where the given PR
+                                                will be stored.
+        """
+        results["content_files"] = {
+            "readme": file_content,
+        }
+
+    def analyse_content_files(
+        self, repository: Repository, prev_knowledge: Dict[str, Any], is_local: bool = False
+    ) -> Dict[str, Any]:
+        """Analyse content files in repository.
+
+        Arguments:
+            repository {Repository} -- currently the PyGithub lib is used because of its functionality
+                                    ogr unfortunatelly did not provide enough to properly analyze issues
+
+            prev_knowledge {Dict[str, Any]} -- previous knowledge stored.
+
+            is_local -- flag to state if the knowledge should be collected locally or on Ceph.
+        """
+        _LOGGER.info("-------------Content Files Analysis-------------")
+
+        # TODO: Extend to all types of files. Currently only README are considered.
+        # TODO: Add all types of README extensions available
+        readme_text = ""
+        for content in ["README.md", "README.rst"]:
+
+            try:
+                readme = repository.get_contents(content)
```

Comment:
what is the intentional use - analysis of a single main readme or all readmes in repository?

Reply:
It can be looked from two sides actually:

1. We can select same file we want to find and look for them, but readme files can be of different format, therefore all formats should be included. (reference: https://pygithub.readthedocs.io/en/latest/examples/Repository.html#get-a-specific-content-file)

2. More like a long term I think but also possible solution right now, get all contents available (e.g. https://pygithub.readthedocs.io/en/latest/examples/Repository.html#get-all-of-the-contents-of-the-root-directory-of-the-repository) and then just retrieve the main readme file not all possible one from all directories for now. We could go inside each dir of the repo as well to discover all files. (in the future when we will analyze also source code will make sense, right now I just wanted to have README.

WDYT?